sharding/indexHashedNodesCoordinator.go

```diff
@@ -882,3 +885,36 @@ func createValidatorInfoFromBody(
 
 	return allValidatorInfo, nil
 }
+
+func toOrderedSlice(validatorMap map[uint32][]Validator, numShardsPlusMeta uint32) [][]Validator {
+	slicedLists := make([][]Validator, numShardsPlusMeta)
+	for shardId, list := range validatorMap {
+		if shardId < numShardsPlusMeta-1 {
+			slicedLists[shardId] = list
+			continue
+		}
+
+		slicedLists[numShardsPlusMeta-1] = list
+	}
+
+	return slicedLists
+}
+
+func toMap(slicedLists [][]Validator, numShardsPlusMeta uint32) map[uint32][]Validator {
+	validatorsMap := make(map[uint32][]Validator)
+
+	for shardId := uint32(0); shardId < numShardsPlusMeta; shardId++ {
+		validatorsList := slicedLists[shardId]
+		if validatorsList == nil {
```

Comment:
this should not happen. all integration tests now have at least 1 node per shard.

Reply:
this is a clear error should be treated

Reply:
undone. not the case any more